File: src/main/java/com/example/movieTheatre/Seat.java
package com.example.movieTheatre;

public class Seat {
    public enum Status {
        ISBEINGOCCUPIED,
        OCCUPIED,
        AVAILABLE,
        NOTBOOKABLE,
        YOURSEAT
    }

    public enum Type {
        STANDARD,
        VIP,
        RECLINER,
        WHEELCHAIRSPACE,
    }

    private Status status;
    private Type type;
    private Person satBy;
    private double priceInPounds;

    public void setType(Type type) {
        this.type = type;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public void setSatBy(Person satBy) {
        this.satBy = satBy;
    }

    public Person getSatBy() {
        return satBy;
    }

    public Status getStatus() {
        return this.status;
    }

    public Type getType() {
        return this.type;
    }

    public double getPriceInPounds() {
        return this.priceInPounds;
    }

    public Seat(Status status, Type type, double priceInPounds) {
        this.status = status;
        this.type = type;
        this.priceInPounds = priceInPounds;
    }

}

File: frontend/components/cinema/seat.js
import Tooltip from "react-tooltip"

const Seat = ({ seat }) => {
	const checkStatus = () => {
		switch (seat.status) {
			case "ISBEINGOCCUPIED":
			case "OCCUPIED":
				return (
					<img
						className='h-10 w-auto'
						src='/seats/OccupiedSeat.svg'
						data-tip={`Occupied Seat`}
						// onClick={(e) => clickASeatHandler(e, seat)}
					/>
				)
			case "NOTBOOKABLE":
				return (
					<img
						className='h-10 w-auto'
						src='/seats/NotBookableSeat.svg'
						data-tip='Not Bookable'
						// onClick={(e) => clickASeatHandler(e, seat)}
					/>
				)
			case "YOURSEAT":
				return (
					<img
						className='h-10 w-auto cursor-pointer'
						src='/seats/Your Seat.svg'
						data-tip='Seat Selected'
						// onClick={(e) => clickASeatHandler(e, seat)}
					/>
				)
			case "AVAILABLE":
			default:
				break
		}

		switch (seat.type) {
			case "STANDARD":
				return (
					<img
						className='h-10 w-auto cursor-pointer'
						src='/seats/standardSeat.svg'
						data-tip='Available Standard Seat'
						// onClick={(e) => clickASeatHandler(e, seat)}
					/>
				)
			case "VIP":
				return (
					<img
						className='h-10 w-auto cursor-pointer'
						src='/seats/VIPSeat.svg'
						data-tip='Available VIP Seat'
						// onClick={(e) => clickASeatHandler(e, seat)}
					/>
				)
			case "RECLINER":
				return (
					<img
						className='h-10 w-auto cursor-pointer'
						src='/seats/TheRecliner.svg'
						data-tip='Available Recliner Seat'
						// onClick={(e) => clickASeatHandler(e, seat)}
					/>
				)
			case "WHEELCHAIRSPACE":
				return (
					<img
						className='h-10 w-auto cursor-pointer'
						src='/seats/WheelchairSpace.svg'
						data-tip='Available Wheelchair seat'
						// onClick={(e) => clickASeatHandler(e, seat)}
					/>
				)
		}
	}

	return checkStatus()
}

export default Seat
